src/main.rs

```diff
@@ -60,10 +61,37 @@ fn read_input<P: AsRef<Path>>(infile: Option<P>) -> Result<String, Box<dyn Error
     Ok(input)
 }
 
-fn write_output<P: AsRef<Path>>(svg: &Element, outfile: Option<P>) -> Result<(), Box<dyn Error>> {
+fn write_output<P: AsRef<Path>>(svg: &Element, outfile: Option<P>, format: Option<String>) -> Result<(), Box<dyn Error>> {
     match outfile {
         Some(filename) => write_to_file(filename.as_ref(), svg)?,
-        None => svg.write_to(&mut std::io::stdout())?,
+        None => {
+            match format {
+                Some(filetype) => {
+                    let mut buffer: Vec<u8> = vec![];
+                    svg.write_to(&mut buffer)?;
+                    let s = std::str::from_utf8(&buffer)?;
+                    let mut fontdb = usvg::fontdb::Database::new();
+                    let font_data = include_bytes!("../resources/Roboto-Bold.ttf").to_vec();
+                    fontdb.load_font_data(font_data);
+                    let tree = usvg::Tree::from_str(
+                    s,
+                    &usvg::Options {
+                        fontdb,
+                        ..usvg::Options::default()
+                    },
+                    )?;
+                    let pixmap_size = tree.svg_node().size.to_screen_size();
+                    let mut pixmap = tiny_skia::Pixmap::new(pixmap_size.width(), pixmap_size.height()).unwrap();
+                    resvg::render(&tree, usvg::FitTo::Original, pixmap.as_mut())
+                        .ok_or(SgfRenderError::PNGRenderFailed)?;
+
+                    io::stdout().write_all(&pixmap.encode_png().unwrap())?;
+                },
+                None => svg.write_to(&mut std::io::stdout())?
+
+            }
+
+        },
```

Comment:
Lets encapsulate all this in a `write_to_stdout` function to make the logic a little cleaner.

Also, there's a bunch of code duplication here. I think it should be possible to write a function that generates the pixmap that gets used in both `write_to_file` and `write_to_stdout` when writing a PNG.